pkg/testutil/testutil.go

```diff
@@ -0,0 +1,329 @@
+		Cluster:  "default-cluster",
+		AuthInfo: "default-user",
+	}
+	authinfos := make(map[string]*clientcmdapi.AuthInfo)
+	authinfos["default-user"] = &clientcmdapi.AuthInfo{
+		ClientCertificateData: restCfg.CertData,
+		ClientKeyData:         restCfg.KeyData,
+	}
+	clientConfig := clientcmdapi.Config{
+		Kind:           "Config",
+		APIVersion:     "v1",
+		Clusters:       clusters,
+		Contexts:       contexts,
+		CurrentContext: "default-context",
+		AuthInfos:      authinfos,
+	}
+
+	d := t.TempDir()
+	kubeconfig, _ := os.CreateTemp(d, "kubeconfig")
+	defer kubeconfig.Close()
+
+	err := clientcmd.WriteToFile(clientConfig, kubeconfig.Name())
+	require.NoError(t, err)
+
+	return kubeconfig.Name()
+}
+
+// Undent removes leading indentation/white-space from given string and returns
+// it as a string. Useful for inlining YAML manifests in Go code. Inline YAML
+// manifests in the Go test files makes it easier to read the test case as
+// opposed to reading verbose-y Go structs.
+//
+// This was copied from https://github.com/jimeh/Undent/blob/main/Undent.go, all
+// credit goes to the author, Jim Myhrberg.
+func Undent(s string) string {
+	const (
+		tab = 9
+		lf  = 10
+		spc = 32
+	)
+
+	if len(s) == 0 {
+		return ""
+	}
+
+	// find smallest indent relative to each line-feed
+	min := 99999999999
+	count := 0
+
+	lfs := make([]int, 0, strings.Count(s, "\n"))
+	if s[0] != lf {
+		lfs = append(lfs, -1)
+	}
+
+	indent := 0
+	for i := 0; i < len(s); i++ {
+		if s[i] == lf {
+			lfs = append(lfs, i)
+			indent = 0
+		} else if indent < min {
+			switch s[i] {
+			case spc, tab:
+				indent++
+			default:
+				if indent > 0 {
+					count++
+				}
+				if indent < min {
+					min = indent
+				}
+			}
+		}
+	}
+
+	// extract each line without indentation
+	out := make([]byte, 0, len(s)-(min*count))
+
+	for i := 0; i < len(lfs); i++ {
+		offset := lfs[i] + 1
+		end := len(s)
+		if i+1 < len(lfs) {
+			end = lfs[i+1] + 1
+		}
+
+		if offset+min <= end {
+			out = append(out, s[offset+min:end]...)
+		} else if offset < end {
+			out = append(out, s[offset:end]...)
+		}
+	}
+
+	return string(out)
+}
+
+// Parses the YAML manifest. Useful for inlining YAML manifests in Go test
+// files, to be used in conjunction with `undent`.
+func Parse(yamlmanifest string) []ctrlruntime.Object {
+	dec := yaml.NewYAMLOrJSONDecoder(strings.NewReader(yamlmanifest), 4096)
+	var objs []ctrlruntime.Object
+	for {
+		obj := &unstructured.Unstructured{}
+		err := dec.Decode(obj)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			panic(err)
+		}
+
+		objs = append(objs, obj)
+	}
+	return objs
+}
+
+type AssertRequest func(t testing.TB, r *http.Request)
+
+func FakeVenafiCloud(t *testing.T) (_ *httptest.Server, _ *x509.Certificate, setAssert func(AssertRequest)) {
+	t.Helper()
+
+	var assertFn AssertRequest
+	assertFnMu := sync.Mutex{}
+	setAssert = func(setAssert AssertRequest) {
+		assertFnMu.Lock()
+		defer assertFnMu.Unlock()
+		assertFn = setAssert
+	}
```

Comment:
I needed to make sure that the request's host was correct, that's why I wrote this clunky code to set an assertion... I'd like to have a clearer way of doing that, or even better: not have to assert the host on the request...